src/database/any.rs

```diff
@@ -312,24 +312,17 @@ impl BatchDatabase for AnyDatabase {
         }
     }
     fn commit_batch(&mut self, batch: Self::Batch) -> Result<(), Error> {
-        // TODO: refactor once `move_ref_pattern` is stable
-        #[allow(irrefutable_let_patterns)]
         match self {
-            AnyDatabase::Memory(db) => {
-                if let AnyBatch::Memory(batch) = batch {
-                    db.commit_batch(batch)
-                } else {
-                    unimplemented!()
-                }
-            }
+            AnyDatabase::Memory(db) => match batch {
+                AnyBatch::Memory(batch) => db.commit_batch(batch),
+                #[cfg(feature = "key-value-db")]
+                _ => unreachable!("Sled batch shouldn't be used with Memory db."),
```

Comment:
Shouldn't this still be `unimplemented!` (with the same message)? since I believe this condition could be reached if `batch` is a `Sled(db)`.

Reply:
Sure thing, changed as suggested. I modified the original commit and force pushed. Is that the typical workflow here or do folks prefer not changing commits after review starts?

Thanks for the review!

Reply:
A  force push is fine, github does a good job of showing just changed files for anyone who's already started a review.

Reply:
Sorry can you do another rebase, there aren't any conflicts but a couple new PRs just got merged.  In particular the PR to changed the CI pipeline required checks from `stable` to `1.50.0` so that commit needs to be in your PR too.

Reply:
No sweat, done.